### PR TITLE
LIV-28

### PR DESCRIPTION
**Pull request by JalenSowell47**
In **config.rb**:
- Added `password_min_length: Devise.password_length.min` to the configuration file to set the minimum length of a password.

In **user.rb**:
- Modified the `password_complexity` method to include a check for the minimum password length.
- Added a class method `min_password_length` to retrieve the minimum password length from the configuration or fallback to the default value provided by Devise.

In the tests **(user_spec.rb)**:
- Added a test to ensure the default password length configuration is set correctly.
- Added tests to validate password length based on the default and custom minimum lengths.
- Created a Config record with a custom minimum password length for testing purposes.

Current questions:
- Are there any edge cases to test for?
- What specifically needs to go into the dev-ops yml files? How often are those ran to update environments?
- Are there any additional tasks or follow-up actions required based on these changes?

### CODE REVIEW

**Review by drewteter — CHANGES_REQUESTED** [review]

File: app/models/user.rb [hunk 1]
```diff
@@ -268,10 +268,24 @@ def downcase_email
...
+# Validates the complexity of the password.
+# This method checks if the password meets the specified complexity requirements.
+# If the password is not present or does not meet the requirements, it adds the
+# appropriate error messages to the `errors` object.
+def password_complexity
+  if password.present?
+    uppercase_complexity = Config.password_required_uppercase <= password.gsub(/([^A-Z])/, '').length
+    lowercase_complexity = Config.password_required_lowercase <= password.gsub(/([^a-z])/, '').length
+    numerical_complexity = Config.password_required_numerical <= password.gsub(/([^0-9])/, '').length
+    special_complexity = Config.password_required_special <= password.gsub(/([^\!\@\#\$\%\^\&\*\(\)\-\=\_\+\;\:\<\>\?\~])/, '').length
```
Comment (drewteter) [inline app/models/user.rb:281]: I'm sorry, I meant for the code I showed to be a rough example. I don't think these are all the possible special characters. You'll have to add whichever ones I left out

**Review by drewteter — APPROVED** [review]
Looks good
